src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
Introducing the same time-based problem again with that change. It will lead to problems much more rarely but it *will* lead to cross process identical random numbers. Any .NET process started within 15ms intervals will have the same random numbers (depending on threading structure)!

Also, the `Random` algorithm is not good. It has problems with near-identical seeds. E.g.: http://stackoverflow.com/questions/36376888/seeding-multiple-random-number-generators. I worry that the seeds x and x plus 15ms might result in correlated numbers.

Is it really that expensive to initialize the Guid generation system? When someone uses random numbers with the big and slow `Random` class, how bad can a Guid generation be?!

This change would be very risky. At least its benefit should be quantified before making it.